### DISCUSSION TITLE
Only retrieving NEAREST within PB2NC

### DISCUSSION THREAD

**Discussion by hertneky** [dtcenter/METplus#2395] [Configuration] [answered]
I've run PB2NC using a time window, but is there also a way to instead only include the NEAREST time for a given station in the pb2nc netcdf output? I looked for the option, but I didn't see anything obvious.

For example:
For one time step at one station, for wind data, you can see three values for the different observation time as follows and to make it so only the closest one to 20220728_06 is in the nc file.
Date StationID Variable OBS_Time Quantity
20220728_06 K1S5 WIND 20220728_053500        0.0000000
20220728_06 K1S5 WIND 20220728_055500        0.0000000
20220728_06 K1S5 WIND 20220728_061500        2.6419690

I realize this capability is available in tools such as point_stat/grid_stat for selecting the nearest from the pb2nc output, but we are not using other tools in this case.

**Comment by JohnHalleyGotway** [comment, marked as answer]
@hertneky I see you have a question about the PB2NC tool. You have a station that reports multiple times in a PrepBUFR file and would like the ability to retain only one of those observations, using the one that is closest to some reference time.

You're correct that Point-Stat does offer this type of observation filtering using the [obs_summary](https://met.readthedocs.io/en/latest/Users_Guide/config_options.html#obs-summary) configuration option. Setting `obs_summary = NEAREST;` tells Point-Stat to use only the point observation that is *closest to the forecast valid time*.

No such functionality currently exists in PB2NC. And the main trouble is the definition of the reference time to which we'd want to be close. It is true that each PrepBUFR file does actually define a reference time. And I think it's usually the mid-point of the retention time window. But some PrepBUFR files contain 3-hours of obs, others contain 6-hours, and others may be hourly. I'm guessing that users would want more flexibility to define the timestamp they'd like to be close to. For example, perhaps you want to process a 6-hour PrepBUFR file and select only those observations closest to the top of each hour.

The existing [time_summary](https://met.readthedocs.io/en/latest/Users_Guide/config_options.html#time-summary) option is supported in PB2NC, as well as the other point pre-processing tools. Its functionality is pretty similar to what you're requesting. Define a time window, find all the obs that fall inside, and summarize them. Here are the existing supported summary types that do what you'd expect:
```
type = [ "min", "max", "range", "mean", "stdev", "median", "p80" ];
```
I can see how we could potentially add another option for "nearest". But there's several little details that should be clarified.


For example, the "max" selects the maximum value of the observations falling within that time range. However, the output timestamp is written based on the time window and *NOT* the input timestamp at which that maximum value occurred.

For "nearest", we'd need to define the reference time that we want to be close to. I'd propose that'd probably be the center of the time window. But the existing code would write out the time window timestamp rather than the actual input timestamp. So perhaps we'd need to add another option to control whether the input timestamp or time window timestamp is written for types min, max, nearest, and percentile values (e.g. p80).

Does this all make sense? Should we add a new MET development issue to enhance the existing point `time_summary` option to add support for `type = nearest`?

**Reply by JohnHalleyGotway** [reply]
@hertneky should I proceed with a new MET GitHub issue to add support for `type = nearest` in the existing `time_summary` option? Or is this not necessarily needed?

**Reply by hertneky** [reply]
@JohnHalleyGotway Thanks for the very thoughtful reply. I understand the enhancement you are proposing. After talking with team members, an enhancement like this is not needed at this time for our project. It could always be an enhancement to consider at a later time, if there if more need.
